### ISSUE TITLE
Custom Prompt Module Doesn't Overriding Checkpoint

### ISSUE THREAD

**Issue by hepengfe** [google-research/prompt-tuning#231]
## Expected Behavior
I changed [prompt](https://github.com/google-research/prompt-tuning/blob/main/prompt_tuning/configs/models/t5_1_1_prompt.gin#L39) to my custom prompt class and load the model from t5x checkpoint. It's expected to override the original prompt module in the checkpoint.

## Actual Behavior
Even though it reads the gin file as reported below
```
I1101 23:22:13.777275 140512320414336 gin_utils.py:65] # Parameters for prompts.CustomPrompt:
I1101 23:22:13.777303 140512320414336 gin_utils.py:65] # ==============================================================================
I1101 23:22:13.777332 140512320414336 gin_utils.py:65] prompts.CustomPrompt.dtype = %ACTIVATION_DTYPE
I1101 23:22:13.777361 140512320414336 gin_utils.py:65] prompts.CustomPrompt.length = %PROMPT_LENGTH
I1101 23:22:13.777390 140512320414336 gin_utils.py:65] prompts.CustomPrompt.prompt_weight_init = \
I1101 23:22:13.777419 140512320414336 gin_utils.py:65]     @prompt_weight_init/linen.initializers.constant()
```
However, it loads original prompt module from the model checkpoint.

## Steps to Reproduce the Problem

1. You can create a dummy custom class that doesn't have to have real content but runnable and syntax correct.
1. Set the prompt config from L40-L43 [here](https://github.com/google-research/prompt-tuning/blob/main/prompt_tuning/configs/models/t5_1_1_prompt.gin#L39)
1. Run the [training script](https://github.com/google-research/prompt-tuning/blob/main/prompt_tuning/scripts/sst2-demo.sh)

## Specifications

- Version: None
- Platform: None

**Comment by hepengfe** [comment]
Do I have to write a class that inherits `Prompt` rather than `nn.Module` to override the original prompt class?

**Comment by blester125** [comment]
> However, it loads original prompt module from the model checkpoint.

I'm not sure I understand your issue here, what do you mean it loads the prompt module from the model checkpoint? Generally there is no prompt module in the checkpoint file. It gets added in and initialized when running the prompt tuning training script.

It looks like your custom prompt is getting picked up and configured by gin, what does it say the value is for `train_prompts.Prompt`?

**Comment by hepengfe** [comment]
Never mind. I think I need to configure the new prompt class in multiple `gin` file to make it work.

* 'prompt_tuning/configs/models/t5_1_1_prompt.gin'
* prompt_tuning/configs/prompts/from_class_labels.gin
* model/t5_small/config.gin

**Comment by blester125** [comment]
Gin sets/overrides values in order, you should be able to configure your custom prompt once as long as it is last, i.e. if you do it in t5_1_1_prompt it gets overridden by the config of from_class_labels. But if you configure it in from class labels (or another gin file which is loaded after) you shouldn't need to update t5_1_1_promot as it will get overridden, so its ok of that part it configured for the default implementation.
